airbyte-protocol/protocol-models/src/main/java/io/airbyte/protocol/models/CatalogHelpers.java

```diff
@@ -384,4 +405,17 @@ static void combineAccumulator(final Map<List<String>, JsonNode> accumulatorLeft
     });
   }
 
+  static Boolean transformBreaksConnection(final ConfiguredAirbyteStream configuredStream, final List<String> fieldName) {
+    final SyncMode syncMode = configuredStream.getSyncMode();
+    if (SyncMode.INCREMENTAL == syncMode && configuredStream.getCursorField().equals(fieldName)) {
+      return true;
+    }
+
+    final DestinationSyncMode destinationSyncMode = configuredStream.getDestinationSyncMode();
+    if (DestinationSyncMode.APPEND_DEDUP == destinationSyncMode && configuredStream.getPrimaryKey().contains(fieldName)) {
```

Comment:
Potential for a subtle bug here as the `equals` of a list is only true when
> two lists are defined to be equal if they contain the same elements in the same order

Is there a potential that these lists contain the same fields in a different order?  Do we need to order these lists _before_ comparing them?

This affects both `configuredStream.getCursorField().equals(fieldName)` and `configuredStream.getPrimaryKey().contains(fieldName)`.

Reply:
I thought there could only ever be one cursorField, but I think changing it to `contains` would fix this. I don't think there's a problem with the second one since it's using `contains` instead of `equals`. Is there something I'm missing about `contains`?

Reply:
Oh I see, fieldName is also a list. I didn't know why this is a list - there is also only ever one item in the fieldName list. I couldn't find examples where there were fieldNames that consisted of a list of multiple strings. I can check with the connectors team on this though.

Reply:
@colesnodgrass it looks like the way we do existing diffs between field names, the order matters. I can't find an example of this, but if one field name is a list of ["field_prefix", "field_name"] and another is ["field_name", "field_prefix"], this would be considered a different field and it would be added to the CatalogDiff

Reply:
Interesting, this may be something we want to explicitly state/enforce somewhere in the code. Or maybe abstract behind a new datatype as there is a potential here for breaking this accidentally.  But this doesn't need to be done as part of this PR.